vision/segmentation/radius_clustering.go

```diff
@@ -118,57 +121,86 @@ func segmentPointCloudObjects(cloud pc.PointCloud, radius float64, nMin int) ([]
 func radiusBasedNearestNeighbors(cloud pc.PointCloud, radius float64) ([]pc.PointCloud, error) {
 	kdt, ok := cloud.(*pc.KDTree)
 	if !ok {
-		// kdt = pc.ToKDTree(cloud)
-		kdt = pc.ToBalancedKDTree(cloud)
+		kdt = pc.ToKDTree(cloud)
 	}
-	// altKdt := pc.ToBalancedKDTree(cloud)
+
+	numGroups := 4
+	subtrees := make([]*pc.KDTree, numGroups)
+
+	subtrees[0] = &pc.KDTree{Tree: &kdtree.Tree{Root: kdt.Tree.Root.Left.Left, Count: 0}}
```

Comment:
These are the very unsafe subtrees I made in an effort to execute the algorithm in parallel